docs/design/2022-11-17-ttl-table.md

```diff
@@ -0,0 +1,347 @@
+  `current_job_id` varchar(64) DEFAULT NULL,
+  `current_job_owner_id` varchar(64) DEFAULT NULL,
+  `current_job_owner_addr` varchar(256) DEFAULT NULL,
+  `current_job_owner_hb_time` timestamp,
+  `current_job_start_time` timestamp NULL DEFAULT NULL,
+  `current_job_ttl_expire` timestamp NULL DEFAULT NULL,
+  `current_job_state` text DEFAULT NULL,
+  `current_job_status` varchar(64) DEFAULT NULL,
+  `current_job_status_update_time` timestamp NULL DEFAULT NULL
+);
+```
+
+It stores some information for each TTL table. The fields prefix `last_job_` present the information of the last job which is successfully executed, and the fields with prefix `current_job_` present the current job which has not been finished yet.
+
+The explanation of the fields:
+
+- `table_id`: The id of the TTL table. If the table is a partitioned table, it stands for the physical table id of each partition.
+- `parent_table_id`: If the current row is a for table partition, it is the table id of its parent table. Otherwise, it equals to `table_id`.
+- `table_statistics`: some statistics of the table.
+- `last_job_id`: The id of the last success job.
+- `last_job_start_time`: The start time of the last job.
+- `last_job_finish_time`: The finish time of the last job.
+- `last_job_ttl_expire`: The expired time used by the last job for TTL works.
+- `last_job_summary`: The summary info for the last job.
+- `current_job_id`: The id of the current job that is not finished. It not only includes the running job, but also includes the job that is failed or cancelled by user.
+- `current_job_owner_id`: The id of the owner (a TiDB node) that runs the job. 
+- `current_job_owner_addr`: The network address of the owner that runs the job.
+- `current_job_owner_hb_time`: The owner of the job updates this field with the current timestamp periodically. If it is not updated for a long time, it means the previous owner is offline and this job should be taken over by other node later. 
+- `current_job_start_time`: The start time of the current job.
+- `current_job_ttl_expire`: The expired time used by the current job for TTL works.
+- `current_job_state`: Some inner state for the current job. It can be used for the job's fail over.
+- `current_job_status`: A enum with one of values: waiting, running, cancelling, cancelled, error
+- `current_job_status_update_time`: The update time of the current status
+
+TTL job for each table runs periodically according to the configuration of the system variable `tidb_ttl_job_run_interval` . For example, if we configure `set @@global.tidb_ttl_job_run_interval='1h'`, the cluster will schedule TTL jobs for each table every one hour to delete expired rows.
+
+If you want to cancel a running job, you can execute a statement like this:
+
+```sql
+ADMIN CANCEL TTL JOB 123456789
+```
+
+In the above example, the status of the TTL job with ID `123456789` will first become `cancelling` and then finally be updated to `cancelled`.
+
+### TTL Job Details
```

Comment:
This part looks like non-transactional delete. Is it possible to reuse it instead of write a new mechanism from scratch?
It doesn't need multiple selects, one select and multiple deletes works. The single select does not necessarily take too much memory since the only thing we need is the boundries of ranges.

Reply:
Yes, it is similar with non-transactional delete, but we want more control in TTL scene. For example, checking schema change and timestamp again to avoid deleting unexpected rows, control the rate limit for all TTL deletions in one node ...